src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIMarsHandle.cs

```diff
@@ -295,15 +295,23 @@ public override uint ReceiveAsync(ref SNIPacket packet)
         /// <summary>
         /// Handle receive error
         /// </summary>
-        public void HandleReceiveError(SNIPacket packet)
+        public void HandleReceiveError(SNIPacket packet, uint sniErrorCode)
         {
             lock (_receivedPacketQueue)
             {
                 _connectionError = SNILoadHandle.SingletonInstance.LastError;
                 _packetEvent.Set();
             }
-
-            ((TdsParserStateObject)_callbackObject).ReadAsyncCallback(PacketHandle.FromManagedPacket(packet), 1);
+            if (sniErrorCode == TdsEnums.SNI_WSAECONNRESET)
```

Comment:
Is this the close or BreakConnection that you are talking about?

Reply:
yes, replied above.